webview-ui/src/__tests__/SettingsTabNavigation.spec.tsx

```diff
@@ -0,0 +1,150 @@
+import React from "react"
+import { render, waitFor } from "@testing-library/react"
+import { vi } from "vitest"
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query"
+
+import App from "../App"
+import TranslationProvider from "../i18n/TranslationContext"
+
+// Mock vscode API
+const mockPostMessage = vi.fn()
+vi.mock("../utils/vscode", () => ({
+	vscode: {
+		postMessage: (message: any) => mockPostMessage(message),
+	},
+}))
+
+// Mock extension state
+const mockExtensionState = {
```

Comment:
I try to avoid this mockExtensionState tests, since you always need to maintain them when you add new things to extension state. Is there a way to write this test without it? If not I might just cut the test.